lib/aescrypt.rb

```diff
@@ -28,19 +28,27 @@
 
 require 'openssl'
 require 'base64'
+require 'securerandom'
+
+require_relative 'migrator'
 
 module AESCrypt
-  def self.encrypt(message, password)
-    Base64.encode64(self.encrypt_data(message.to_s.strip, self.key_digest(password), nil, "AES-256-CBC"))
+  def self.encrypt(message, password, salt = nil, iv = nil)
+    iv ||= SecureRandom.bytes 16
+    salt, key = self.key_digest(password, salt)
```

Comment:
See my comment below; an HMAC (with salt) is unnecessary for generating cipher keys.

The point of a key derivation function is to expand or contract the bit length of the key to something that's the block length of cipher. If slightly different keys yield vastly different results from the key derivation function, that's also a benefit. So running the key through such a function is great!

But the use of an HMAC requires you to deal with the salt; a simple SHA2-256 for AES-256 would be sufficient and no salt would be required. You can hash for a few iterations much like the `OpenSSL::PKCS5.pbkdf2_hmac` function does, but that's about all that's required.